semana-22/case-1/src/business/UserBusiness.ts

```diff
@@ -0,0 +1,41 @@
+import { UserDataBase } from "../data/UserDataBase";
+import { MissingFields } from "../error/MissingFields";
+import { UserInput, UserInsert } from "../modal/User";
+import { IdGenerator } from "../services/IdGenerator";
+
+export class UserBusiness {
+    signUpBusiness = async(input: UserInput) => {
+        if(!input.name || 
+            !input.email || 
+            !input.data_passeio ||
+            !input.duracao ||
+            !input.latitude ||
+            !input.longitude ||
+            !input.horario_inicio ||
+            !input.horario_termino
+            ){
+                throw new MissingFields()
+            }
+
+        // if((input.horario_termino - input.horario_inicio) !== 30 || 
+        //     (input.horario_termino - input.horario_inicio) !== 60) {
+        //     throw new Error("Insira um horario com intervalo de 30 ou 60 minutos")
+        // }
+        // // pergunta para o plantão: Como ajustar a validação do horário?
```

Comment:
Pense na perspectiva do front e do back. Como alguém agendaria um passeio?

Um exemplo:
- a API recebe somente o horário de início e a duração
- valida se o horário de início está no futuro
- valida se a duração é realmente 30 ou 60
- o horário de término é calculado automaticamente